functions/src/definitions/onInstanceCreate.js

```diff
@@ -68,6 +69,28 @@ exports.onInstanceCreate = functions
           error
         )
       }
+    } else if (data?.type === "image") {
+      //TODO: IF IMAGE, UPDATE storageURL, minetype, text etc. Also update typesense db
+      parentMessageRef.update({
+        latestInstance: snap.ref,
+        caption: data.caption,
+      })
+      if (data?.embedding && data?.text) {
+        const updateObj = {
+          id: snap.ref.path.replace(/\//g, "_"), //typesense id can't seem to take /
+          message: data.text,
+          captionHash: data.captionHash ? data.captionHash : "__NULL__",
+          embedding: data.embedding,
+        }
+        try {
+          await insertOne(updateObj, CollectionTypes.Instances)
```

Comment:
repeated logic, whole block could be inside 1 try/catch actually

Reply:
Done